### PR TITLE
Update daml 2.10's upgrade documentation with latest changes

### PR DESCRIPTION
**Pull request by paulbrauner-da**
- Update the doc to mention LF 1.17 instead of 1.16
- Update the doc with the latest restrictions wrt to interfaces

### CODE REVIEW

File: docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst [hunk 1]
```diff
@@ -2023,3 +2033,123 @@ have been written using the template on the left with ``n = 0``.
...
+.. code:: daml
+
+    data IView = IView { i : Int }
+  
+    interface I where
+      viewtype IView
+      m : Int
+ 
+Then below the template on the right is a valid upgrade of the template on the
+left.
```
Comment (carrielaben-da) [inline docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst:2059]:
```suggestion
In that case, the template on the right below is a valid upgrade of the template on the
left.
```

**Review by carrielaben-da — APPROVED** [review]
One small suggestion for clarity.

File: docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst [hunk 2]
```diff
@@ -163,7 +163,7 @@ To prevent unexpected behavior, this feature enforces a unique package name and
 uploaded to a participant node.
 This closes a loophole where the participant node allowed multiple DARs with
 the same package name and version. For backward compatibility, this
-restriction only applies for packages compiled with LF >= 1.16. If LF <
+restriction only applies for packages compiled with LF >= 1.17. If LF <
 1.15 is used, there can be several packages with the same name and
```
Comment (dylant-da) [inline docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst:167]: The following line probably needs updating to be "if LF < 1.16", or a note needs to be made of the behaviour in 1.16

**Review by dylant-da — APPROVED** [review]
Aside from comments, looks good - are we burning 1.16 completely? I didn't realize

File: docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst (hunk 2 above)
Comment (paulbrauner-da) [reply docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst:167]: As discussed offline, everything is as if 1.16 had never existed, so it's fine for its behavior to be unspecified.

**Comment by samuel-williams-da** [comment]
I believe these docs also exist in 3.x, do they need the same changes?

File: docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst [hunk 3]
```diff
@@ -1916,11 +1932,11 @@ JSON API Server
...
 
 Template IDs may still be used with a package ID, however,
-for packages built as LF 1.16 or greater, the package may also be
+for packages built as LF 1.17 or greater, the package may also be
 identified by name. That is to say, for upgradable packages a template ID can have
 the form ``#<package-name>:<module-name>:<template-name>``, and this is
 resolved to corresponding templates from all packages which share this
-name, and are built at 1.16 or above. For packages built at LF 1.15 or
+name, and are built at 1.17 or above. For packages built at LF 1.15 or
 lower, the templates are not identifiable via a package name, and a
```
Comment (remyhaemmerle-da) [inline docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst:1940]:
```suggestion
name, and are built at 1.17 or above. For packages built at LF 1.15,
the templates are not identifiable via a package name, and a
```

**Review by remyhaemmerle-da — APPROVED** [review]
Nice

File: docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst [hunk 4]
```diff
@@ -1630,37 +1626,57 @@ DARs. They should both succeed again:
...
   DAR upload succeeded.
   > cd ../../v2/my-pkg
   > daml build
-  > daml sandbox upload-dar --port 6865
+  > daml ledger upload-dar --port 6865
+  ...
+  Uploading .daml/dist/my-pkg-2.0.0.dar to localhost:6865
+  upload-dar did not succeed: ... Implementation of interface ...:MyIface:HasValue by template IOU appears in package that is being upgraded, but does not appear in this package.
+
+Packages with new versions cannot add an interface instance to an existing
```
Comment (yangzai) [inline docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst:1661]: Hi @paulbrauner-da , under "Best Practices" > "Remove Retroactive Instances", it still states that "retroactive interface instances should be moved to newer versions of templates, such that changes to the instance warrants a new version of the template".

Doesn't it conflict with the new restriction here that "packages with new versions cannot add an interface instance to an existing template"?

Comment (paulbrauner-da) [reply docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst:1661]: Thank you for pointing this out, we'll fix the inconsistency.

**Pull request merged**